### DISCUSSION TITLE
Lose progression on restart

### DISCUSSION THREAD

**Discussion by Phoenix-Game** [mbround18/valheim-docker#606] [Ideas]
Any chance to make server save all progression before autorestart or simple restart, each time server restart we lose about 20-30 minutes of progression sometimes less but it's critical when no admin online to forcefully save progression before restart.

**Comment by spannerman79** [comment]
On a clean shutdown (`SIGINT`) the server correctlys saves.

What shutdown signal are you issuing before shutdown?

Using docker-compose or docker run?

Provide that with obvious personal info edited/replaced in a **code block** please - makes it easier to read instead of a wall of text.

**Reply by mbround18** [reply]
@Phoenix-Game hmmm thats odd and shouldnt be happening, when the auto restart job comes it it triggers a sigint, which @spannerman79 has mentioned saves the server before shutdown. Is the container getting killed in some way?

**Reply by Phoenix-Game** [reply]
Looks like restart works like it should work but anyway we got few days in a row losses of progression on our map, and it's about 20 minutes or 15, a lot of players lost to many resources.

**Reply by Phoenix-Game** [reply]
I'm using docker-compose and to shutdown or restart the container I use web which is portainer. But problems with losses comes all the time when server is autorestarting. So I don't use anything to autorestart it just Env with cron at fixed time.

**Reply by spannerman79** [reply]
`stop_signal: SIGINT` needs to be added to your compose while using portainer.

**Reply by spannerman79** [reply]
@mbround18 not an issue with the image, its just using portainer and not taking the time to actually read portainers docs.

@Phoenix-Game when you use docker compose - ie `docker-compose down` etc the stop signal sent is actually `SIGINT`.  Its only portainer that sends `SIGTERM` which causes the issue.

**Reply by Phoenix-Game** [reply]
> causes

Yes but what about autorestart it's not about protainer and it doesn't have anything with portainer. The issue most of the time comes with autorestart

**Reply by spannerman79** [reply]
Sigh....

Lets get into this;

>I'm using docker-compose and to shutdown or restart the container I use web which is **portainer.**

So **yes** you **are** using portainer.

 >The issue most of the time comes with autorestart

Doesn't apply, Odin's auto restart sends `SIGINT` so its not Odin or using `SCHEDULED_RESTART` as **that actually uses `SIGINT`**.  I've actually monited it myself personally to ensure that it works correctly to what the linux dev over at Iron Gate implimented as per requested by other people running servers on linux and native linux users themselves.

**Reply by Phoenix-Game** [reply]
I can test it now, I've got an SCHEDULED_RESTART in 30 minutes and if I'll get losses of progression again I'll let you know.

**Reply by Phoenix-Game** [reply]
Just now after SCHEDULED_RESTART I have lost about 15 minutes of my progression in the world.

**Reply by mbround18** [reply]
@Phoenix-Game after work today I'll spin up a test server to validate this feature

**Reply by Phoenix-Game** [reply]
@mbround18 Is there any news? I still have this problem even with new changes in docker-compose file: stop_signal: SIGINT
 Nothing changes. I also installed everything on a new server with just docker, without portainer or something else, and I've got the same issue.

**Reply by mbround18** [reply]
@Phoenix-Game well you caught me when I was moving my entire family cross country lol. I just finished the move and unpacking, gimme a day or so to unpack my equipment and I'll give it a whirl. Have you tried without any mods? When you use "odin stop" inside the container and then "odin start" does it save your changes?

**Reply by Phoenix-Game** [reply]
@mbround18 No prob, take your time. I'll try it without mods and I'll let you know

**Reply by mbround18** [reply]
@Phoenix-Game apologies its been a really rough whirl wind. got the pc setup finally and downloading game now to validate. should have a conclusion tomorrow

**Reply by mbround18** [reply]
@Phoenix-Game I have tried the auto restart, it does save my progress. do you potentially have a mod what might be interfering with the save process?

**Reply by spannerman79** [reply]
I'm going to assume that the issue isn't related to the (docker) image but to mods as its been **20 days** since any further updates.
